RAGworkshop/README.md

```diff
@@ -0,0 +1,426 @@
+`security.protocol=SASL_SSL`  
+`sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username=XXXXXX password='XXXXXXX';`  
+`sasl.mechanism=PLAIN`
+```
+
+2. Analytical connection details - these are to be used with analytical tooling for more complex, SQL based queries.   
+   
+```
+`SET SESSION streambased_connection = '{`  
+  `"bootstrap.servers": "pkc-l6wr6.europe-west2.gcp.confluent.cloud:9092",`  
+  `"security.protocol": "SASL_SSL",`  
+  `"sasl.jaas.config": "org.apache.kafka.common.security.plain.PlainLoginModule required username=''XXXXXXXXXXXX'' password=''XXXXXXXXX'';",`  
+  `"sasl.mechanism": "PLAIN",`  
+  `"schema.registry.url":"https://psrc-571d82.europe-west2.gcp.confluent.cloud",`  
+  `"basic.auth.credentials.source":"USER_INFO",`  
+  `"basic.auth.user.info":"XXXXXX"`  
+`}';`
+```
+Copy the analytical connection details for later use.
+
+**Step 2: Explore the data using analytical tools**
+
+Note: This section uses Streambased A.S.K., a free, hosted analytical tool for Kafka based on the open source Trino project. 
+
+Streambased A.S.K. offers a JDBC/ODBC/SQL Alchemy connection and so can be integrate with most analytical tools. We love the simplicity of [SquirrelSQL](https://squirrel-sql.sourceforge.io/) but feel free to follow the guides below to connect the tool of your choice.
+
+[https://www.streambased.io/tutorial-guides](https://www.streambased.io/tutorial-guides)
+
+Remember the analytical connection details we collected from RDx? We can apply these with a single command in our SQL tool to repoint A.S.K. at our Kafka data.
+
+```
+`SET SESSION streambased_connection = '{`  
+  `"bootstrap.servers": "pkc-l6wr6.europe-west2.gcp.confluent.cloud:9092",`  
+  `"security.protocol": "SASL_SSL",`  
+  `"sasl.jaas.config": "org.apache.kafka.common.security.plain.PlainLoginModule required username=''XXXXXXXXXXXX'' password=''XXXXXXXXX'';",`  
+  `"sasl.mechanism": "PLAIN",`  
+  `"schema.registry.url":"https://psrc-571d82.europe-west2.gcp.confluent.cloud",`  
+  `"basic.auth.credentials.source":"USER_INFO",`  
+  `"basic.auth.user.info":"XXXXXX"`  
+`}';`
+```
+
+
+1. Listing Topics \- To list all available topics, simply use the following command:
+
+`show tables;`
+
+![image1](images/image1.png)
+
+2. Viewing message structure \- as with listing, the topic/table duality means that viewing the structure of a message is as simple as describing the table:
+
+`DESCRIBE yt_gaming_subtitles;`
+
+![image2](images/image2.png)
+
+Note: any column beginning with an underscore is an internal column that reflects Kafka specific metadata. These columns can be used in queries just like any other. 
+
+3. Fetching some data \- now we can explore the Kafka data in the same way that we would any other SQL based resource:
+
+`SELECT * FROM yt_gaming_subtitles LIMIT 100;`
+
+
+**CHAPTER 2: Realtime and RAG**
+
+**Setting up Jupyter notebook:-**  
```

Comment:
no need, it's already there at localhost:8888

Reply:
In this chapter please open the Jupyter notebook provided by our docker using localhost:8888